bundles/org.openhab.binding.iaqualink/src/main/java/org/openhab/binding/iaqualink/internal/handler/IAqualinkHandler.java

```diff
@@ -0,0 +1,450 @@
+     * @param httpClient
+     */
+    public IAqualinkHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        client = new IAqualinkClient(httpClient);
+    }
+
+    @Override
+    public void initialize() {
+        configure();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        clearPolling();
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        // clear our cached value so the new channel gets updated on the next poll
+        stateMap.remove(channelUID.getId());
+    }
+
+    @SuppressWarnings("null") // stateMap.remove can return null, which confuses the compiler
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand channel: {} command: {}", channelUID, command);
+
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            logger.warn("Controller is NOT ONLINE and is not responding to commands");
+            return;
+        }
+
+        clearPolling();
+
+        String channelName = channelUID.getIdWithoutGroup();
+        // remove the current state to ensure we send an update
+        State oldState = stateMap.remove(channelUID.getAsString());
+        try {
+            if (command instanceof RefreshType) {
+                stateMap.clear();
+            } else if (channelName.startsWith("onetouch_")) {
+                if (oldState == null || !oldState.toString().equals(command.toString())) {
+                    client.oneTouchCommand(serialNumber, sessionId, "set_" + channelName);
+                }
+            } else if (channelName.endsWith("heater") || channelName.endsWith("pump")) {
+                if (oldState == null || !oldState.toString().equals(command.toString())) {
+                    client.homeScreenCommand(serialNumber, sessionId, "set_" + channelName);
+                }
+            } else if (channelName.startsWith("aux_")) {
+                if (command instanceof OnOffType) {
+                    if (oldState == null || !oldState.toString().equals(command.toString())) {
+                        client.auxCommand(serialNumber, sessionId, "set_" + channelName);
+                    }
+                }
+                if (command instanceof DecimalType) {
+                    client.auxCommand(serialNumber, sessionId, "set_" + channelName,
+                            ((DecimalType) command).intValue());
+                }
+            } else if ("spa_set_point".equals(channelName) && command instanceof DecimalType) {
+                client.setSpaTemp(serialNumber, sessionId, ((DecimalType) command).intValue());
+            } else if ("pool_set_point".equals(channelName) && command instanceof DecimalType) {
+                client.setPoolTemp(serialNumber, sessionId, ((DecimalType) command).intValue());
+            }
+            initPolling(COMMAND_REFRESH_SECONDS);
+        } catch (InterruptedException e) {
```

Comment:
If we receive this InterruptedException, the interrupted flag was cleared and we most likely need to reset the flag:

http://www.yegor256.com/2015/10/20/interrupted-exception.html
https://www.ibm.com/developerworks/library/j-jtp05236/

Reply:
hmm, so the only time i have seen this exception thrown is when i call `localFuture.cancel(true);` , where i am explicitly telling the future to interrupt the current executor, which is by design.  The only other time i can think this would happen is during an unloading of the binding, is there another situation where this could happen?

Reply:
I refactored my poll logic to check if the local executor is still valid instead of relying on the interrupted exception throw by the jetty client.